src/main/java/team/asd/service/ProductService.java

```diff
@@ -0,0 +1,89 @@
+package team.asd.service;
+
+import lombok.NonNull;
+import org.apache.commons.collections4.CollectionUtils;
+import team.asd.constants.ProductState;
+import team.asd.entities.IsProduct;
+import team.asd.exceptions.WrongProductException;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+public class ProductService implements IsProductService {
+	@Override
+	public @NonNull List<String> defineProductNames(List<IsProduct> productList) throws WrongProductException {
+		if (CollectionUtils.isEmpty(productList))
+			return Collections.emptyList();
+
+		for (IsProduct product : productList) {
+			if (product == null || product.getName() == null)
+				throw new WrongProductException("There is product which is null or without name");
+		}
+
+		return productList.stream()
+				.map(IsProduct::getName)
+				.collect(Collectors.toList());
+	}
+
+	@Override
+	public List<IsProduct> defineProductsWithCreatedState(List<IsProduct> productList) {
+		return CollectionUtils.isEmpty(productList) ?
+				Collections.emptyList() :
+				productList.stream()
+						.filter(product -> isProductValid(product) && product.getState()
+								.equals(ProductState.Created))
```

Comment:
You should use ProductState.Created.equals(product.getState()) to avoid possible NullPointerException